### PR TITLE
fixed tests requiring gradients

### PR DESCRIPTION
**Pull request by puzzleshark**
Do to recent [refactor](https://github.com/PennyLaneAI/pennylane/commit/bc822b40d97c11fc59570f96d4280374cee639ac) of PennyLane, default behaviour of `requires_grad` some scalar values need to be explicitly set as autograd tensors, with `requires_grad=True`.

### CODE REVIEW

**Review by antalszava — APPROVED** [review]
Looks good! :tada: Thank you!

For some reason, we don't seem to be running the device tests that we run in the plugin test matrix. :thinking: Do the tests pass now well locally with these changes?

**Comment by antalszava** [comment]
@Jaybsoni it looks like the only changes affecting the logic were made to the test files. Therefore, we would not need to release this plugin.

**Comment by Jaybsoni** [comment]
> @Jaybsoni it looks like the only changes affecting the logic were made to the test files. Therefore, we would not need to release this plugin.

Sounds good ! 
So we only need to version bump the Qiskit plugin

**Comment by antalszava** [comment]
[sc-14395]
